gateway_test.go

```diff
@@ -1029,3 +1030,48 @@ func TestProxyUserAgent(t *testing.T) {
 		}
 	}
 }
+
+func buildAndLoadAPI(apiGens ...func(spec *APISpec)) {
+	for i, gen := range apiGens {
+		spec := &APISpec{APIDefinition: &apidef.APIDefinition{}}
+		json.Unmarshal([]byte(sampleAPI), spec)
```

Comment:
Unmarshal into `spec.APIDefinition`, just in case

Reply:
Not sure what you mean here

Reply:
`sampleAPI` is an `APIDefinition`, not an `APISpec`. One is embedded into the other, so in practice it's probably fine, but it could cause issues if any field names ever conflict or if the JSON string ever includes fields directly from `APISpec`.

So I'm saying that you should do `json.Unmarshal([]byte(sampleAPI), spec.APIDefinition)`, to make sure none of those are possible.

Reply:
👌